### PR TITLE
Fix `prevent-pr-commit-link-loss`

### LINKED ISSUE #3222: `prevent-pr-commit-link-loss` doesn't respect URL hash
E.g. for the following link:

```
https://github.com/sindresorhus/refined-github/pull/3205/commits/1da152b3f8c51dd72d8ae6ad9cc96e0c2d8716f5#diff-932095cc3c0dff00495b4c392d78f0afR60
```

… it should rather suggest the following:

```diff
- [https://github.com/sindresorhus/refined-github/pull/3205/commits/1da152b3f8c51dd72d8ae6ad9cc96e0c2d8716f5 ](https://github.com/sindresorhus/refined-github/pull/3205/commits/1da152b3f8c51dd72d8ae6ad9cc96e0c2d8716f5)#diff-932095cc3c0dff00495b4c392d78f0afR60
+ [https://github.com/sindresorhus/refined-github/pull/3205/commits/1da152b3f8c51dd72d8ae6ad9cc96e0c2d8716f5#diff-932095cc3c0dff00495b4c392d78f0afR60](https://github.com/sindresorhus/refined-github/pull/3205/commits/1da152b3f8c51dd72d8ae6ad9cc96e0c2d8716f5#diff-932095cc3c0dff00495b4c392d78f0afR60)
```

In short:

```diff
- [pr_url/commits/commit_ref ](pr_url/commits/commit_ref)#hash
+ [pr_url/commits/commit_ref#hash](pr_url/commits/commit_ref#hash)
```

Also, it adds an unnecessary space after the commit ref.

### PR DESCRIPTION
**Pull request by fregante**
## Blinking message

Fixes https://github.com/sindresorhus/refined-github/pull/3085#issuecomment-636536392

1. Paste `https://github.com/sindresorhus/refined-github/pull/3085/commits/a7b1a8bd85c230ff9ff8e407ca8f6d3646c90eb6` in a field
2. Type something else
3. The message should appear and _not_ blink at every keypress

## URLs with hashes

Fixes #3222

1. Paste `https://github.com/sindresorhus/refined-github/pull/3205/commits/1da152b3f8c51dd72d8ae6ad9cc96e0c2d8716f5#diff-932095cc3c0dff00495b4c392d78f0afR60` in a field
2. Click `Fix link`
3. The resulting link should include the `#diff- etc cetera` part

## Debounces

This should avoid slowing down typing long comments (regardless of broken links)

1. Enter a broken URL
2. Start typing right away
3. The message should not appear until you stop typing (but the debounce timer is short)

## Shorten URL

Since we're creating a custom link, we might as well make it look good for everyone

1. Paste `https://github.com/sindresorhus/refined-github/pull/3085/commits/a7b1a8bd85c230ff9ff8e407ca8f6d3646c90eb6` in a field
2. Click `Fix link`
3. The link should appear as 

	```md
	[`1da152b` (#3205)](https://github.com/sindresorhus/refined-github/pull/3205/commits/1da152b3f8c51dd72d8ae6ad9cc96e0c2d8716f5) 
	```

Note: the hash doesn't appear in this shortened version... but it's no biggie IMHO

### CODE REVIEW

**Comment by fregante** [comment]
I'm stuck on the regex. I can no longer make it match the URL _unless it ends with `)`_

**Comment by fregante** [comment]
Successful test with [`06d10f3` (#3223)](https://github.com/sindresorhus/refined-github/pull/3223/commits/06d10f3658196f612980d864baf32e35147f1352)

File: source/github-helpers/index.ts [hunk 1]
```diff
@@ -97,8 +97,14 @@ export function getLatestVersionTag(tags: string[]): string {
 }
 
 const escapeRegex = (string: string) => string.replace(/[\\^$.*+?()[\]{}|]/g, '\\$&');
-export const prCommitRegex = new RegExp(`\\b${escapeRegex(location.origin)}[/][^/]+[/][^/]+[/]pull[/]\\d+[/]commits[/][0-9a-f]{7,40}\\b(?! \\]|\\))`, 'gi');
+const prCommitPathnameRegex = /[/][^/]+[/][^/]+[/]pull[/](\d+)[/]commits[/]([\da-f]{7})[\da-f]{33}(?:#[\w-]+)?\b/; // eslint-disable-line unicorn/better-regex
```
Comment (FloEdelmann) [inline source/github-helpers/index.ts:100]: Why do you use `[/]` instead of `\/`?

Comment (fregante) [reply source/github-helpers/index.ts:100]: I generally find them easier to follow. Example, how long does it take to understand what this regex matches?

```js
/^\/\/\*\??/
```

<details>
<summary><code>[] equivalent</code></summary>


```
/^[/][/][*][?]?/
```
matches

```
'//*?' or '//*'
```
</details>

**Pull request merged**